interfaces/apparmor/backend.go

```diff
@@ -176,30 +176,38 @@ func (b *Backend) Initialize() error {
 	return nil
 }
 
-// snapConfineFromCoreProfile returns the apparmor profile for snap-confine in the given core snap.
-func snapConfineFromCoreProfile(coreInfo *snap.Info) (dir, glob string, content map[string]*osutil.FileState, err error) {
+// snapConfineFromSnapProfile returns the apparmor profile for
+// snap-confine in the given core/snapd snap.
+func snapConfineFromSnapProfile(info *snap.Info) (dir, glob string, content map[string]*osutil.FileState, err error) {
 	// Find the vanilla apparmor profile for snap-confine as present in the given core snap.
 
 	// We must test the ".real" suffix first, this is a workaround for
 	// https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=858004
-	vanillaProfilePath := filepath.Join(coreInfo.MountDir(), "/etc/apparmor.d/usr.lib.snapd.snap-confine.real")
+	vanillaProfilePath := filepath.Join(info.MountDir(), "/etc/apparmor.d/usr.lib.snapd.snap-confine.real")
 	vanillaProfileText, err := ioutil.ReadFile(vanillaProfilePath)
 	if os.IsNotExist(err) {
-		vanillaProfilePath = filepath.Join(coreInfo.MountDir(), "/etc/apparmor.d/usr.lib.snapd.snap-confine")
+		vanillaProfilePath = filepath.Join(info.MountDir(), "/etc/apparmor.d/usr.lib.snapd.snap-confine")
 		vanillaProfileText, err = ioutil.ReadFile(vanillaProfilePath)
 	}
 	if err != nil {
 		return "", "", nil, fmt.Errorf("cannot open apparmor profile for vanilla snap-confine: %s", err)
 	}
 
 	// Replace the path to vanilla snap-confine with the path to the mounted snap-confine from core.
-	snapConfineInCore := filepath.Join(coreInfo.MountDir(), "usr/lib/snapd/snap-confine")
+	snapConfineInCore := filepath.Join(info.MountDir(), "usr/lib/snapd/snap-confine")
 	patchedProfileText := bytes.Replace(
 		vanillaProfileText, []byte("/usr/lib/snapd/snap-confine"), []byte(snapConfineInCore), -1)
-	// /snap/core/111/usr/lib/snapd/snap-confine -> snap.core.111.usr.lib.snapd.snap-confine
-	patchedProfileName := strings.Replace(snapConfineInCore[1:], "/", ".", -1)
-	// snap.core.111.usr.lib.snapd.snap-confine -> snap.core.*.usr.lib.snapd.snap-confine
-	patchedProfileGlob := strings.Replace(patchedProfileName, "."+coreInfo.Revision.String()+".", ".*.", 1)
+
+	// We need to add a uniqe prefix that can never collide with a
+	// snap on the system. Using "snap-confine.*" is similar to
+	// "snap-update-ns.*" that is already used there
+	//
+	// So
+	//   /snap/core/111/usr/lib/snapd/snap-confine
+	// becomes
+	//   snap-confine.snap.core.111.usr.lib.snapd.snap-confine
+	patchedProfileName := "snap-confine." + strings.Replace(snapConfineInCore[1:], "/", ".", -1)
```

Comment:
I'm not sure I understand why you're doing the replace manually for the first /, but it doesn't make a difference :-)

Reply:
@chipaca this is similar (but different) to systemd-escape --path. We don't want the leading slash to become the leading dot :)

Reply:
Since we are already using `/var/lib/snapd/apparmor/profiles/` we can do less hard-coding of the full path of the profile executable like what is typical in `/etc/apparmor.d`. I'd personally just use `snap-confine.{core,snapd}.$rev`.

Please apply this change if you are going to make any revisions of this code.